server/broadcast.go

```diff
@@ -57,16 +76,31 @@ func (pi *packetInfo) broadcastAll(msgs []*message.Signed) {
 	defer pi.tracker.mutex.RUnlock()
 
 	sender := pi.tracker.connections[pi.conn]
+	if sender == nil {
+		sender = pi.tracker.findPlayerInPoolSnapshots(pi.conn)
+		if debugMode {
+			fmt.Printf(logDirectMessage+"Found %v for missing broadcast message sender\n", sender)
+		}
+	}
 
-	// If the user has disconnected, then no need to send
-	// the broadcast.
+	// If the user has disconnected and cannot be found, there is no way to broadcast
 	if sender == nil {
+		if debugMode {
+			fmt.Printf(logBroadcast+"Ignoring message from %s because player is missing and pool cannot be identified\n", getIP(pi.conn))
+		}
 		return
 	}
```

Comment:
Although the send-after-disconnect logic works for a direct message, it doesn't work for broadcasts where we have no way to verify which pool to send to.

Therefore I added a pool search - try to find a disconnected user by the connection signature in all of the frozen pools. If we find one there, then we can proceed with a broadcast. If we do not, there is no way to do a broadcast.

Reply:
This could be quite expensive... will give it some thought.

Reply:
resolving since the offending code is not visible any more.